commodore/inventory/parameters.py

```diff
@@ -0,0 +1,229 @@
+
+    @property
+    def region(self):
+        return self._region
+
+
+class InventoryParameters:
+    def __init__(self, inv, node, invfacts: InventoryFacts):
+        self._inventory = inv
+        self._node = node
+        self._distribution = invfacts.distribution
+        self._cloud = invfacts.cloud
+        self._region = invfacts.region
+
+    @property
+    def distribution(self):
+        return self._distribution
+
+    @property
+    def cloud(self):
+        return self._cloud
+
+    @property
+    def region(self):
+        return self._region
+
+    def parameters(self, param: Optional[str] = None):
+        params = self._inventory["nodes"][self._node]["parameters"]
+        if param is not None:
+            params = params.get(component_parameters_key(param), {})
+
+        return params
+
+    @property
+    def applications(self):
+        return self._inventory["nodes"][self._node]["applications"]
+
+
+class DefaultsFact(Enum):
+    DISTRIBUTION = "distribution"
+    CLOUD = "cloud"
+    REGION = "region"
+
+
+class InventoryFactory:
+    def __init__(self, work_dir: Path, global_dir: Path):
+        self._global_dir = global_dir
+        self._inventory = Inventory(work_dir=work_dir)
+        self._distributions = self._find_values(DefaultsFact.DISTRIBUTION)
+        self._clouds = self._find_values(DefaultsFact.CLOUD)
+        self._cloud_regions: Dict[str, Iterable[str]] = {}
+        for cloud in self._clouds:
+            r = self._find_values(DefaultsFact.REGION, cloud=cloud)
+            self._cloud_regions[cloud] = [it for it in r if it != "params"]
+
+    @property
+    def directory(self) -> Path:
+        return self._inventory.inventory_dir
+
+    @property
+    def classes_dir(self) -> Path:
+        return self._inventory.classes_dir
+
+    @property
+    def targets_dir(self) -> Path:
+        return self._inventory.targets_dir
+
+    @property
+    def global_dir(self) -> Path:
+        return self._global_dir
+
+    def _reclass_config(self) -> Dict:
+        return {
+            "storage_type": "yaml_fs",
+            "inventory_base_uri": str(self.directory.absolute()),
+            "nodes_uri": str(self.targets_dir.absolute()),
+            "classes_uri": str(self.classes_dir.absolute()),
+            "compose_node_name": False,
+            "allow_none_override": True,
+            "ignore_class_notfound": True,
+        }
+
+    def reclass(self, invfacts: InventoryFacts) -> InventoryParameters:
+        distribution = invfacts.distribution
+        cloud = invfacts.cloud
+        region = invfacts.region
+        if distribution is None:
+            distribution = "x-fake-distribution"
+        if cloud is None:
+            cloud = "x-fake-cloud"
+        if region is None:
+            region = "x-fake-region"
+
+        c: Dict[str, Any] = {
+            "id": "c-bar",
+            "tenant": "t-foo",
+            "displayName": "Foo Inc. Bar cluster",
+            "facts": {
```

Comment:
I assume that we will add more and more dynamic facts. Having to touch this each time we do so, would be a nuisance. Instead of handling the known facts explicitly, I suggest to do this in a more generic way.